From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Teroy Goph <159986878+gteroy@users.noreply.github.com>
Date: Sat, 27 Jul 2024 17:28:49 +0800
Subject: [PATCH] 6764: Fix:debugger-agent-gdb:remote make sure brk.commands is
 not none

make sure `brk.commands` is not none

When in debugging in gdb remote, `brk.commands` would be None thus Exception is thrown which then hang the Ghidra debugger
---
 .../Debug/Debugger-agent-gdb/src/main/py/src/ghidragdb/hooks.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Ghidra/Debug/Debugger-agent-gdb/src/main/py/src/ghidragdb/hooks.py b/Ghidra/Debug/Debugger-agent-gdb/src/main/py/src/ghidragdb/hooks.py
index beffc33f0d..c0495c39c0 100644
--- a/Ghidra/Debug/Debugger-agent-gdb/src/main/py/src/ghidragdb/hooks.py
+++ b/Ghidra/Debug/Debugger-agent-gdb/src/main/py/src/ghidragdb/hooks.py
@@ -312,6 +312,8 @@ def check_for_continue(event):
             return True
         for brk in event.breakpoints:
             if hasattr(brk, 'commands'):
+                if brk.commands is None:
+                    return False
                 for cmd in brk.commands:
                     if cmd == 'c' or cmd.startswith('cont'):
                         HOOK_STATE.in_break_w_cont = True
-- 
2.45.1

